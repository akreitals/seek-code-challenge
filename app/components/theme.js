const maxContainerWidth = '940px';

const emoji = '"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"';
const font = `Averta, -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Helvetica, Arial, sans-serif, ${emoji}`;
const mono = 'SFMono-Regular,"Roboto Mono",Menlo,monospace';

const fontSizes = [12, 14, 16, 18, 24, 32, 48, 64, 72, 96];
const space = [0, 4, 8, 16, 32, 64, 128];
const regular = 400;
const bold = 600;

const greys = {
    black: '#1c1c1c',
    charcoal: '#404040',
    midGrayDark: '#747474',
    midGrayMedium: '#898989',
    midGray: '#a8a8a8',
    midGrayLight: '#d6d6d6',
    grayMedium: '#e1e1e1',
    grayLight: '#eeeeee',
    grayLightest: '#f7f7f7',
    offWhite: '#fbfbfb',
    white: '#ffffff'
};

const brand = {
    blueDarker: '#071d40',
    blueDark: '#042763',
    blue: '#0d3880',
    blueLight: '#184da8',
    blueLighter: '#2765cf',
    pink: '#e60278',
    green: '#157e00',
    greenLight: '#5bc252',
    purple: '#9556b7',
    teal: '#00a5ad',
    orange: '#f57c00',
    yellow: '#ffc600',
    yellowLight: '#fffce3',

    primary: '#e60278',
    accent: '#9556b7',
    success: '#157e00',
    info: '#2765cf',
    warning: '#f57c00',
    error: '#e60278'
};

export const colors = {
    ...greys,
    ...brand
};

export const radius = '2px';

// boxShadows
export const boxShadows = [
    `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
];

const theme = {
    colors,
    font,
    mono,
    fontSizes,
    space,
    regular,
    bold,
    maxContainerWidth,
    radius,
    boxShadows
};

export default theme;
