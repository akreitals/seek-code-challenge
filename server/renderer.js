import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
