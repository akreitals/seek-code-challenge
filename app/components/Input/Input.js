import styled from 'styled-components';
import { space, width, borders } from 'styled-system';
import PropTypes from 'prop-types';
import theme, { radius, colors } from '../theme';

const Input = styled('input')`
    appearance: none;
    display: block;
    vertical-align: middle;
    max-width: 32rem;
    color: ${colors.charcoal};
    min-height: 36px;
    font-family: inherit;
    line-height: ${3 * theme.space[3]}px};
    font-size: ${theme.fontSizes[3]}px;
    background-color: transparent;
    border-radius: ${radius};
    border: 1px solid ${colors.midGrayLight};
    padding-top: 14px;
    padding-bottom: 14px;
    padding-left: 12px;
    padding-right: 12px;
    margin: 0;
    width: 100%;
    ::placeholder {
        color: ${colors.midGrayLight};
    }
    ::-ms-clear {
        display: none;
    }
    &:focus {
        outline: none;
        z-index: 1;
        box-shadow: 0 0 0 2px rgba(30, 144, 255, 0.75);

        &::-moz-focus-inner {
            border: 0;
        }
    }
    ${width} ${space} ${borders};
`;

Input.displayName = 'Input';

Input.propTypes = {
    id: PropTypes.string.isRequired,
    color: PropTypes.string,
    ...width.PropTypes,
    ...space.PropTypes,
    ...borders.PropTypes
};

Input.defaultProps = {
    theme,
    w: 1,
    m: 0,
    py: 0,
    px: '15px'
};

export default Input;
