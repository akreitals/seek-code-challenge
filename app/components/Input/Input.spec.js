import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from './Input';
import theme from '../theme';

describe('Input', () => {
    test('should render a <Input /> component', () => {
        const json = renderer.create(<Input id="renders" />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('should render with custom spacing', () => {
        const json = renderer.create(<Input p={4} m={4} id="space" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule(`padding ${theme.space[4]}px`);
        expect(json).toHaveStyleRule(`margin ${theme.space[4]}px`);
    });
});
