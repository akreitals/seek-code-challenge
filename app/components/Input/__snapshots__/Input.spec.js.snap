// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render a <Input /> component 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  color: #404040;
  min-height: 36px;
  font-family: inherit;
  line-height: 48px;
  font-size: 18px;
  background-color: transparent;
  border-radius: 2px;
  border: 1px solid #d6d6d6;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  width: 100%;
  margin: 0px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 0px;
  padding-bottom: 0px;
}

.c0::-webkit-input-placeholder {
  color: #d6d6d6;
}

.c0::-moz-placeholder {
  color: #d6d6d6;
}

.c0:-ms-input-placeholder {
  color: #d6d6d6;
}

.c0::placeholder {
  color: #d6d6d6;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 2px rgba(30,144,255,0.75);
}

.c0:focus::-moz-focus-inner {
  border: 0;
}

<input
  className="c0"
  id="renders"
/>
`;

exports[`Input should render with custom spacing 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  color: #404040;
  min-height: 36px;
  font-family: inherit;
  line-height: 48px;
  font-size: 18px;
  background-color: transparent;
  border-radius: 2px;
  border: 1px solid #d6d6d6;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  width: 100%;
  margin: 32px;
  padding: 32px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 0px;
  padding-bottom: 0px;
}

.c0::-webkit-input-placeholder {
  color: #d6d6d6;
}

.c0::-moz-placeholder {
  color: #d6d6d6;
}

.c0:-ms-input-placeholder {
  color: #d6d6d6;
}

.c0::placeholder {
  color: #d6d6d6;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 2px rgba(30,144,255,0.75);
}

.c0:focus::-moz-focus-inner {
  border: 0;
}

<input
  className="c0"
  id="space"
/>
`;
